src/plib/library/DomainConnect.php

```diff
@@ -137,18 +258,66 @@ public function init()
 
         \pm_Log::debug("Domain {$this->domain->getDisplayName()} is resolved to {$resolvedIp}, but expected " . join(' or ', $hostingIps));
 
-        $serviceId = \pm_Config::get('webServiceId');
         $properties = ['ip' => reset($hostingIps)];
         if ($this->domainHasParent()) {
             $properties['host'] = DomainDns::relativeHost($this->getRootDomainName(), $this->domain->getName());
-            $properties['groupId'] = 'web';
+            $properties['groupId'] = \pm_Config::get('webServiceGroupId');
+        }
+
+        $url = $this->getApplyTemplateUrl($this->serviceId, $properties);
+
+        $this->domain->setSetting('connectable', 1);
+        $this->domain->setSetting('configureUrl', $url);
+
+        if (isset($this->getData()->width, $this->getData()->height)) {
+            $this->domain->setSetting('windowOptionWidth', $this->getData()->width);
+            $this->domain->setSetting('windowOptionHeight', $this->getData()->height);
+        }
+    }
+
+
+    public function initService()
```

Comment:
you "general" code for an "abstract" service is still focused on the mail service.